File: Job Sequencing Problem[ 17 - April - 2023]/Solution.java
int[] JobScheduling(Job arr[], int n)
    {
        boolean[] jobsDone = new boolean[n];
        Arrays.fill(jobsDone, false);
        Arrays.sort(arr, (a, b) -> b.profit - a.profit);
        int cnt = 0, profit = 0;
        for (Job job : arr) {
            for (int i = Math.min(job.deadline - 1, n - 1); i >= 0; i--) {
                if (!jobsDone[i]) {
                    cnt++;
                    profit += job.profit;
                    jobsDone[i] = true;
                    break;
                }
            }
        }
        return new int[]{cnt, profit};
    }

File: Job Sequencing Problem[ 17 - April - 2023]/solution.py
#   >> PYTHON CODE <<


def JobScheduling(self,Jobs,n):
    jobsDone=[False]*n
    
    Jobs.sort(key=lambda x: -x.profit)
    
    
    cnt,profit=0,0
    
    for job in Jobs:
        for i in range(min(job.deadline-1,n-1),-1,-1):
            if jobsDone[i]==False:
                cnt+=1
                profit+=job.profit
                jobsDone[i]=True;
                break
    
    return [cnt,profit]